src/engraving/libmscore/navigate.cpp

```diff
@@ -965,4 +966,29 @@ EngravingItem* Score::prevElement()
     }
     return score()->firstElement();
 }
+
+//---------------------------------------------------------
+//   prevLyrics
+//    - find the lyric (if any) before this one (not including lines)
+//    - currently used to determine the first lyric of a melisma
+//---------------------------------------------------------
+
+Lyrics* Lyrics::prevLyrics() const
```

Comment:
This is not exactly what I meant :) You should declare a function in navigate.h (similar to prevChordRest):

`extern Lyrics* prevLyrics(const Lyrics* lyrics);
`

Then use this new function in TextBase::endEdit(EditData& ed):

```
 if (isLyrics()) {
     Lyrics* prevLyrics = prevLyrics(toLyrics(this));
     if (prevLyrics) {
         ...
     }
 }


```

Reply:
For the record, why do we want to do this? There are no other Lyrics methods in navigate.cpp, so I am wondering what the point of all of this is?

Reply:
Because it can iterate implicitly over the whole score. And in general, has low cohesion in this class (it does not depend on anything within this class, and can easily be moved out of it and take it as an argument). We want engraving elements to be as simple as possible